develop/plone/sessions/cookies.rst

```diff
@@ -2,71 +2,73 @@
+Setting cookies is done on the response.
+Each cookie name there can be a new or an existing one.
+When sending the request, the information is turned into a properly formatted `Set-Cookie` header.
 
-Modifying HTTP response cookies
-===============================
+.. code-block:: python
 
-You might want to tune up or clean cookies after some other part of Plone code has set them.
-You can do this in :doc:`post-publication event handler </develop/plone/serving/http_request_and_response>`.
+    self.request.response.setCookie("cookie_name", "cookie_value")
 
-Example ``cleancookies.py`` (needs ZCML subscriber registration too)::
+A cookie may have `cookie-attributes <https://en.wikipedia.org/wiki/HTTP_cookie#Cookie_attributes>`_ set.
+Those are passed as keyword arguments to ``setCookie``.
+Turning a cookie into a non-session cookie requires a expires='date' keyword,
+limiting the cookie to a path is a path='/somepath' keyword to the `setCookie()` method.
+The usual browser cookie rules apply here.
 
-    """
+.. code-block:: python
 
-        Clean I18N cookies from non-HTML responses so that e.g. Image
-        content, which has language set, and is cross-linked across page,
-        don't inadvertiately change the langauge.
+    self.request.response.setCookie(
+        "cookie_name",
+        "cookie_value",
+        quoted=False,  # default is True
+        attribute_on_cookie="attribute value",
+        another_attribute="another attribute value",
+    )
 
-    """
 
-    from zope.interface import Interface
-    from zope.component import adapter
-    from plone.postpublicationhook.interfaces import IAfterPublicationEvent
+Clearing cookies
+================
 
+If a cookie needs to be removed, the browser has to be told to expire it.
+The ``expireCookie`` method does this.
+It always set ``max_age`` to ``0`` and the ``expires`` date to the past.
+Additional the cookie value will be set to ``deleted``.
 
-    @adapter(Interface, IAfterPublicationEvent)
-    def clean_language(object, event):
-        """ Clean up cookies after HTTPResponse object has been constructed completely.
-
-        Post-publication handler.
-        """
-        request = event.request
+.. code-block:: python
 
-        #print "%s %s" % (request["URL"], request.response.cookies)
+    self.request.response.expireCookie("cookie_name")
 
-        # All non-HTML payloads
-        if not request.response.headers["content-type"].startswith("text/html"):
-            # Rip-off I18N_language cookie
-            if "I18N_LANGUAGE" in request.response.cookies:
-                print "Cleaned up cookie for %s" % request["URL"]
-                del request.response.cookies["I18N_LANGUAGE"]
+The method allows additional attributes to be passed as kwargs, similar to ``setCookie``.
+Except ``max_age``, ``expires`` and ``value`` all is allowed, like `path` or other cookie attributes.
```

Comment:
I don't understand this sentence together with the previous. Can you clarify the intent? It seems to say that the following kwargs are reserved keywords, but it's not clear. Also use the Oxford comma and double backticks.

```
Except ``max_age``, ``expires``, and ``value`` all is allowed, like ``path`` or other cookie attributes.
```

Reply:
These two sentences together on lines 70-71 still don't make sense to me. Can you clarify?